### PR TITLE
fix(theme): finalize themes with design system

### PR DESCRIPTION
**Pull request by nnixaa**
BREAKING CHANGES:
- the following sass mixins are removed: `nb-radial-gradient`, `nb-right-gradient`
- common theme variables changes:

renamed:
  font-main -> font-family-primary
  font-secondary -> font-family-secondary
  radius -> border-radius
  color-bg -> background-basic-color-1,
  color-bg-active -> background-basic-color-2,
  color-fg -> text-basic-color,
  color-fg-heading -> text-basic-color,
  color-fg-text -> text-hint-color,
  separator -> divider-color
  scrollbar-fg -> scrollbar-color,
  scrollbar-bg -> scrollbar-background-color,
  scrollbar-thumb-radius -> scrollbar-width

removed:
  color-fg-highlight: #40dc7e,

  font-weight-thin: 200,
  font-weight-light: 300,
  font-weight-normal: 400,
  font-weight-bolder: 500,
  font-weight-bold: 600,
  font-weight-ultra-bold: 800,

  base-font-size: 16px,

  font-size-xlg: 1.25rem,
  font-size-lg: 1.125rem,
  font-size: 1rem,
  font-size-sm: 0.875rem,
  font-size-xs: 0.75rem,

  padding: 1.25rem,
  margin: 1.5rem,
  line-height: 1.25,

  separator: #ebeef2,// TODO: move to constants
  social-color-facebook: #3b5998,
  social-color-twitter: #55acee,
  social-color-google: #dd4b39,
  social-color-linkedin: #0177b5,
  social-color-github: #6b6b6b,
  social-color-stackoverflow: #2f96e8,
  social-color-dribble: #f26798,
  social-color-behance: #0093fa,

  radial-gradient: none,
  linear-gradient: none,

### CODE REVIEW

File: src/framework/theme/styles/themes/_default.scss [hunk 1]
```diff
@@ -8,57 +8,14 @@
...
-  scrollbar-thumb-radius: 2.5px,
-
-  radial-gradient: none,
-  linear-gradient: none,
-
-  /* Eva theme variables */
+
+  /**
+   * Sass map contains a list of all Theme variables and also their mappings into Component variables
+   * A theme contains of a list of colors, backgrounds, borders, text styles and supporting variables.
```
Comment (yggg) [inline src/framework/theme/styles/themes/_default.scss]: A theme ~contain~ consists

File: src/framework/theme/styles/themes/_default.scss [hunk 2]
```diff
@@ -110,6 +67,8 @@ $theme: (
   color-danger-800: #94124e,
   color-danger-900: #700940,
 
+  /* Basic colors - for backgrounds and borders and texts */
```
Comment (yggg) [inline src/framework/theme/styles/themes/_default.scss]: >backgrounds and borders and texts

Is it ok to use 'and' with every word?

File: src/framework/theme/styles/themes/_default.scss [hunk 3]
```diff
@@ -707,43 +672,43 @@ $theme: (
...
-  smart-table-header-font-size: font-size,
-  smart-table-header-font-weight: font-weight-bold,
-  smart-table-header-line-height: line-height,
-  smart-table-header-fg: color-fg-heading,
-  smart-table-header-bg: color-bg,
+  smart-table-header-font-family: text-paragraph-font-family,
+  smart-table-header-font-size: text-paragraph-font-size,
+  smart-table-header-font-weight: text-paragraph-font-weight,
+  smart-table-header-line-height: text-paragraph-line-height,
+  smart-table-header-fg: text-color,
```
Comment (yggg) [inline src/framework/theme/styles/themes/_default.scss]: `text-color` -> `text-basic-color`?
There is more usages below

File: src/framework/theme/styles/themes/_cosmic.scss [hunk 4]
```diff
@@ -32,18 +32,18 @@ $theme: (
...
+  background-basic-color-3: color-basic-700,
+  background-basic-color-4: color-basic-800,
+
+  border-basic-color-1: color-basic-500,
+  border-basic-color-2: color-basic-600,
+  border-basic-color-3: color-basic-700,
+  border-basic-color-4: color-basic-800,
+  border-basic-color-5: color-basic-900,
+
+  text-default-color: color-basic-200,
```
Comment (yggg) [inline src/framework/theme/styles/themes/_cosmic.scss]: `text-basic-color`?

File: src/framework/theme/styles/themes/_corporate.scss [hunk 5]
```diff
@@ -9,33 +9,21 @@
 @import 'default';
 
 $theme: (
-  separator: #cdd5dc,
   radius: 0.17rem,
```
Comment (yggg) [inline src/framework/theme/styles/themes/_corporate.scss]: `border-radius`?

**Comment by nnixaa** [comment]
we also need to remove `divider-color: border-default-color-2,` from dark theme

**Comment by yggg** [comment]
Remove or change to `border-basic-color-2`?

**Pull request merged**